### PR TITLE
Economizer Refactor

### PR DESCRIPTION
**Pull request by ChristopherKnight**
Creating the request to pull the updated economizer refactor into the repository

### CODE REVIEW

**Comment by rlutes** [comment]
In new_data_message the current time needs to be localized.  Although this does not effect the economizer for consistency all times input into the diagnostics should be localized if possible.

File: pnnl/EconomizerRCxAgent/economizer/constants.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/constants.py]: Please add file header

File: pnnl/EconomizerRCxAgent/economizer/diagnostics/EconCorrectlyOff.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
+"""
+Copyright (c) 2017, Battelle Memorial Institute
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/diagnostics/EconCorrectlyOff.py]: Change the year to 2020

File: pnnl/EconomizerRCxAgent/economizer/diagnostics/ExcessOutsideAir.py [hunk 3]
```diff
@@ -1,346 +1,264 @@
...
+        """ Calculate the impact the temperature values have
+        desired_oaf: float
+
+        returns float
+        """
+        ei = 0.0
+        energy_calc = [
+            (1.08 * spd * self.cfm * (m - (o * desired_oaf + (r * (1.0 - desired_oaf))))) / (1000.0 * self.eer)
+            for m, o, r, spd in zip(self.mat_values, self.oat_values, self.rat_values, self.fan_spd_values)
+            if (m - (o * desired_oaf + (r * (1.0 - desired_oaf)))) > 0
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/diagnostics/ExcessOutsideAir.py]: more meaningful variables will be helpful in understanding what's going on here.

File: pnnl/EconomizerRCxAgent/economizer/diagnostics/InsufficientOutsideAir.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+        avg_oaf = max(0.0, min(100.0, avg_oaf))
+        for sensitivity, threshold in self.ventilation_oaf_threshold.items():
+            if self.desired_oaf - avg_oaf > threshold:
+                msg = "{}: Insufficient OA is being provided for ventilation - sensitivity: {}".format(constants.ECON5, sensitivity)
+                result = 43.1
+            else:
+                msg = "{}: The calculated OAF was within acceptable limits - sensitivity: {}".format(constants.ECON5, sensitivity)
+                result = 40.0
+            _log.info(msg)
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/diagnostics/InsufficientOutsideAir.py:157]: Remove if unnecessary

File: pnnl/EconomizerRCxAgent/economizer/economizer_agent.py [hunk 5]
```diff
@@ -0,0 +1,560 @@
+
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/economizer_agent.py]: Needs a file header

File: pnnl/EconomizerRCxAgent/economizer/economizer_agent.py [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+        #diagnostics
+        self.temp_sensor = None
+        self.econ_correctly_on = None
+        self.econ_correctly_off = None
+        self.excess_outside_air = None
+        self.insufficient_outside_air = None
+
+        #start reading all the class configs and check them
+        self.read_config(config_path)
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/economizer_agent.py:169]: Please add config store support.

File: pnnl/EconomizerRCxAgent/economizer/economizer_agent.py [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+                    self.device_list.append(topics.DEVICES_VALUE(campus=self.campus, building=self.building, unit=u, path=sd, point="all"))
+
+    def read_argument_config(self, config_path):
+        """read all the config arguments section
+        no return
+        """
+        config = utils.load_config(config_path)
+        self.arguments = config.get("arguments", {})
+
+        self.econ_hl_temp = self.read_argument("econ_hl_temp", 65.0)
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/economizer_agent.py:257]: If you add config store support, default values can be set in default config dictionary

File: pnnl/EconomizerRCxAgent/economizer/economizer_agent.py [hunk 8]
```diff
@@ -0,0 +1,560 @@
...
+
+        return float
+        return Bool/int
+        """
+        if self.device_type == "ahu":
+            clg_vlv_pos = mean(self.cooling_data)
+            cool_call = True if clg_vlv_pos > self.cooling_enabled_threshold else False
+        elif self.device_type == "rtu":
+            cool_call = int(max(self.cooling_data))
+
```
Comment (shwethanidd) [inline pnnl/EconomizerRCxAgent/economizer/economizer_agent.py:630]: If self.device_type is neither "ahu" nor "rtu" then cool_call is not defined and throw error later

**Review by shwethanidd — CHANGES_REQUESTED** [review]
The code seems to be pretty good. Major comments are: 
1. Readme describing about the agent and it's usage is missing. This would be beneficial for external users.
2. Econimizer agent needs to support config store.
3. Some of the equation variables have one syllable names (hard to read the code)
4. File headers are missing for some files

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**